homeassistant/components/homewizard/__init__.py

```diff
@@ -64,12 +64,28 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.async_create_task(hass.config_entries.async_remove(old_config_entry_id))
 
     # Create coordinator
-    coordinator = Coordinator(hass, entry.data[CONF_IP_ADDRESS])
+    coordinator = Coordinator(hass, entry.entry_id, entry.data[CONF_IP_ADDRESS])
     try:
         await coordinator.async_config_entry_first_refresh()
+
     except ConfigEntryNotReady:
-        await coordinator.api.close()
-        raise
+
+        if not coordinator.api_disabled:
+            await coordinator.api.close()
+            raise
+
+    # Setup entry
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    if coordinator.api_disabled:
+        await coordinator.async_listen_for_api_enabled()
```

Comment:
Why is this necessary? Isn't this automatically retried when HA retries setup and it calls `async_config_entry_first_refresh` ?

Reply:
I did not know that HA retries the setup without user interaction when setup failed?

I added a listener to have at lease one listener to the coordinator. The platforms are not loaded when the first refresh was unsuccessful, so they do not listen.

Reply:
Never mind, I read now that it should do that by raising ConfigEntryNotReady. 

Should we trigger the reauth-flow manually and then call NotReady?

Reply:
Yes

Reply:
Actually, shouldn't it just be a repair issue in that case ?

Reply:
Maybe not for discussion here, but why is the reauth flow not a repair issue?

Do you want me to take a look at the repair in this PR or should we finish this one up and open a follow-up PR?

Reply:
> Maybe not for discussion here, but why is the reauth flow not a repair issue?

That is one of the goals that hasn't been completed yet.

Reply:
I've implemented the reauth logic triggered from `ConfigEntryNotReady` and it works really well. When the API is re-enabled the config entry is reloaded automatically. 

> Actually, shouldn't it just be a repair issue in that case ?
@balloon, Do you want to keep it like this or that we create a repair flow? In my opinion it feels a bit weird to create a 'custom' repair issue if we can use the native re-auth flow.

Reply:
Any thoughts on this?